### DISCUSSION TITLE
Vorrei creare un DDT (delivery note) contestualmente alla creazione di una PROFORMA

### DISCUSSION THREAD

**Discussion by stefanosilvestrinimia** [fattureincloud/api#331] [Q&A] [answered]
ciao, 
vorrei creare un DDT contestualmente alla chiamata API per creare una PROFORMA. 

Utilizzo SDK PHP e so che dovrei popolare i campi che iniziano con dn_ (dn_number, dn_date ...).

Soluzioni?

grazie!

**Comment by valmoz** [comment]
Buongiorno @stefanosilvestrinimia ,
le invio un po' di documentazione che potrebbe esserle utile:

- Documentazione della [PHP SDK](https://developers.fattureincloud.it/docs/sdks/php-sdk)
- Readme [PHP SDK](https://github.com/fattureincloud/fattureincloud-php-sdk/tree/master)
- [Quickstart per PHP SDK](https://developers.fattureincloud.it/docs/quickstarts/php-sdk-quickstart)
- Metodo [Create Issued Document](https://github.com/fattureincloud/fattureincloud-php-sdk/blob/master/docs/Api/IssuedDocumentsApi.md#createissueddocument)
- [Modello Issued Document](https://github.com/fattureincloud/fattureincloud-php-sdk/blob/master/docs/Model/IssuedDocument.md), in cui sono presenti i campi "dn_"
- [Guida alla creazione di un Issued Document](https://developers.fattureincloud.it/docs/guides/invoice-creation) (è basata sulla fattura, ma riadattabile agli altri tipi)

**Reply by stefanosilvestrinimia** [reply]
grazie @valmoz .

Abbiamo già creato le proforme con Issued Document (e quindi abbiamo già studiato tutta la documentazione PHP SDK, etc). Ti chiedo se puoi solo indicarmi come inserire un DDT contestualmente alla creazione di un issued Document. Grazie

**Reply by emajo** [reply]
Ciao @stefanosilvestrinimia i campi del ddt dalla nostra piattaforma web sono i seguenti:
![campi](https://github.com/fattureincloud/api/assets/48432054/b74c0b32-cc4c-4e96-9bde-94e605760f34)
che si traducono per quanto riguarda l'api in questi campi
![campi2](https://github.com/fattureincloud/api/assets/48432054/12c9ef45-6f69-4211-94e0-dc5ce3c15cce)

esempio PHP:

```
$doc = new IssuedDocument;
$doc->setType(IssuedDocumentType::PROFORMA);
$doc->setDnDate(new DateTime("2022-06-28"));
$doc->setDnNumber(7);
...
```

**Reply by stefanosilvestrinimia** [reply]
tutto chiarissimo grazie. Ho aggiunto tutti i campi necessari:

> $invoice->setDnDate(new DateTime("2023-06-28"));
# $invoice->setDnNumber(7);
$invoice->setDnAiPackagesNumber('1');
$invoice->setDnAiWeight('1');
$invoice->setDnAiCausal('casuale casuale');
$invoice->setDnAiTransporter('DHL DHL');
$invoice->setDnAiNotes('notes notes');
$invoice->setDnAiDestination('milano');

la proforma viene correttamente creata ma il DDT non viene creato (nonostante i campi). Idee del perchè?

grazie ancora

**Reply by stefanosilvestrinimia** [reply]
credo bastasse aggiungere:

> $invoice->setDeliveryNote(true);

**Reply by stefanosilvestrinimia** [reply]
@emajo sai dirmi come posso NON settare questo flag all'interno delle IssuedDocumentItemsListItem? 

![Schermata 2023-06-28 alle 17 46 49](https://github.com/fattureincloud/api/assets/114735413/506649a2-013c-4537-bd78-fe3648d1ddda)

**Reply by emajo** [reply, marked as answer]
@stefanosilvestrinimia per escludere il prodotto dal ddt è necessario settare il campo `in_dn` presente all'interno di `items_list` a false
